runtime/src/in_mem_accounts_index.rs

```diff
@@ -347,20 +347,20 @@ impl<T: IndexValue> InMemAccountsIndex<T> {
     pub fn update_key_if_exists(
         &self,
         pubkey: &Pubkey,
-        new_value: &AccountMapEntry<T>,
+        new_value: PreAllocatedAccountMapEntry<T>,
         reclaims: &mut SlotList<T>,
         previous_slot_entry_was_cached: bool,
-    ) -> bool {
+    ) -> (bool, Option<PreAllocatedAccountMapEntry<T>>) {
```

Comment:
I'm guessing the CI 'checks' step failed due to callers of this function needing to handle the different return type.